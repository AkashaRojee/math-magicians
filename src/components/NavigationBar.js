import { Link } from 'react-router-dom';
import preval from 'preval.macro';

const NavigationBar = () => {
  const routeFiles = preval`
    const fs = require('fs');
    const files = fs.readdirSync('src/routes');
    module.exports = files;
  `;

  const extractRoutes = (routeFile) => {
    const routeName = routeFile.slice(routeFile.indexOf('-') + 1, -3);
    const path = (routeName === 'Home' ? '' : routeName.toLowerCase());
    return [routeName, path];
  };

  let index = 0;

  return (
    <nav>
      {routeFiles.map((routeFile) => {
        const [routeName, path] = extractRoutes(routeFile);
        index += 1;

        return (
          <Link
            to={`/${path}`}
            key={index}
          >
            {routeName}
          </Link>
        );
      })}
    </nav>
  );
};

export default NavigationBar;
