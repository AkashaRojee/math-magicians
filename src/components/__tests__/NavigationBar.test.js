import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import NavigationBar from '../NavigationBar';

test('NavigationBar renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <NavigationBar />
      </BrowserRouter>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
