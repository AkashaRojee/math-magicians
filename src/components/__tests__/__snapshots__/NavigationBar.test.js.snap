// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly 1`] = `
<nav>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <a
    href="/calculator"
    onClick={[Function]}
  >
    Calculator
  </a>
  <a
    href="/quote"
    onClick={[Function]}
  >
    Quote
  </a>
</nav>
`;
